aten/src/ATen/native/special_functions/detail/numeric.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
```

Comment:
@albanD specialized complex type promotion (see promote.h too). This is mostly useful for special functions but could probably be moved to c10 at some point.

Reply:
There's 
```
using value_t = typename c10::scalar_value_type<scalar_t>::type;
```
typically used to extract dtype from complex

Reply:
Awesome! I'll swap this.